api/subscriptions/subscriptions.go

```diff
@@ -387,15 +339,76 @@ func (s *Subscriptions) Close() {
 	s.wg.Wait()
 }
 
+func (s *Subscriptions) websocket(readerFunc func(http.ResponseWriter, *http.Request) (msgReader, error)) utils.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) error {
+		s.wg.Add(1)
+		defer s.wg.Done()
+
+		// Call the provided reader function
+		reader, err := readerFunc(w, req)
+		if err != nil {
+			return err
+		}
+
+		// Ensure cleanup when the connection closes
+		defer func() {
+			if closer, ok := reader.(interface{ Close() }); ok {
```

Comment:
`MsgReader` doesn't required to implement a `closer` interface, this code block is not required.

Reply:
Oh yeah. Leftover from the handlePendingTransactions refactor. Good catch! 🙏